object-common.go

```diff
@@ -166,26 +167,19 @@ func putObjectCommon(storage StorageAPI, bucket string, object string, size int6
 	if err != nil {
 		return "", err
 	}
+	err = storage.DeleteFile(bucket, object)
+	if err != nil && err != errFileNotFound {
+		return "", err
+	}
+	err = storage.RenameFile(minioMetaBucket, tempObj, bucket, object)
+	if err != nil {
+		return "", err
+	}
```

Comment:
It is unsafe to do this delete and rename. Let posix Rename do this at the lowest level once. Infact it should move original to tmp, rename and then remove tmp.